File: backend/src/main/java/com/seatarrangement/business/service/SeatService.java
package com.seatarrangement.business.service;

import com.seatarrangement.common.entity.Seat;
import com.seatarrangement.common.entity.Employee;
import java.util.List;

public interface SeatService {
    List<Seat> getSeatsByFloor(Integer floorNo);
    void assignSeat(Integer floorSeatSeq, String empId);
    void clearSeat(Integer floorSeatSeq);
    List<Employee> getAllEmployees();
}

File: frontend/src/services/seatService.ts
import axios from 'axios';
import type { Seat, Employee } from '../types';

const API_BASE_URL = 'http://localhost:8080/api';

const apiClient = axios.create({
    baseURL: API_BASE_URL,
    headers: {
        'Content-Type': 'application/json',
    },
});

export default {
    getSeatsByFloor(floorNo: number) {
        return apiClient.get<Seat[]>(`/floors/${floorNo}/seats`);
    },
    getEmployees() {
        return apiClient.get<Employee[]>('/employees');
    },
    assignSeat(floorSeatSeq: number, empId: string) {
        return apiClient.post('/seats/assign', { floorSeatSeq, empId });
    },
    clearSeat(floorSeatSeq: number) {
        return apiClient.post('/seats/clear', { floorSeatSeq });
    },
};
